bus-mapping/src/circuit_input_builder.rs

```diff
@@ -392,49 +515,28 @@ impl<'a> CircuitInputStateRef<'a> {
             });
         }
 
-        // Return from a {CREATE, CREATE2} with a failure, via RETURN
-        // TODO: if not self.tx_ctx.is_root and self.tx_ctx.is_create {
-        // https://github.com/appliedzkp/zkevm-circuits/issues/137
-        if step.op == OpcodeId::RETURN && next_result == Word::zero() {
-            let offset = step.stack.nth_last(0).unwrap();
-            let length = step.stack.nth_last(1).unwrap();
-            if length > Word::from(0x6000) {
-                return Some(ExecError::MaxCodeSizeExceeded);
-            } else if length > Word::zero()
-                && !step.memory.0.is_empty()
-                && step.memory.0.get(offset.low_u64() as usize) == Some(&0xef)
-            {
-                return Some(ExecError::InvalidCode);
-            } else if Word::from(200) * length > Word::from(step.gas.0) {
-                return Some(ExecError::CodeStoreOutOfGas);
-            } else {
-                panic!("Cannot figure out RETURN error in {:?}", step);
-            }
-        }
-
         // The *CALL* code was not executed
         let next_pc = next_step.map(|s| s.pc.0).unwrap_or(1);
-        if [
-            OpcodeId::CALL,
-            OpcodeId::CALLCODE,
-            OpcodeId::DELEGATECALL,
-            OpcodeId::STATICCALL,
-            OpcodeId::CREATE,
-            OpcodeId::CREATE2,
-        ]
-        .contains(&step.op)
-            && next_result == Word::zero()
+        if matches!(
+            step.op,
+            OpcodeId::CALL
+                | OpcodeId::CALLCODE
+                | OpcodeId::DELEGATECALL
+                | OpcodeId::STATICCALL
+                | OpcodeId::CREATE
+                | OpcodeId::CREATE2
+        ) && next_result == Word::zero()
             && next_pc != 0
         {
             if step.depth == 1025 {
                 return Some(ExecError::Depth);
             }
             // TODO: address_collision
             // https://github.com/appliedzkp/zkevm-circuits/issues/180
-            if step.op == OpcodeId::CREATE || step.op == OpcodeId::CREATE2 {
+            if matches!(step.op, OpcodeId::CREATE | OpcodeId::CREATE2) {
```

Comment:
Why not using `is_create`? Since it's implemented within this PR already.

Reply:
Unfortunately `is_create` is defined for `CallType`, and here we have `OpcodeId`.